src/server_manager/electron_app/windows/electron_builder_signing_plugin.cjs

```diff
@@ -23,6 +23,13 @@
 async function electronBuilderEntryPoint(configuration) {
   // CommonJS module is required, ES6 module is not supported by electron-builder
   const {signWindowsExecutable} = await import('./sign_windows_executable.mjs');
+  if (configuration.hash === 'sha1') {
+    // SHA-1 signatures are retired and should be replaced by SHA-2, Windows 7
+    // users are required to upgrade to Windows 7 SP1 (or install the standalone
+    // update KB4474419)
+    console.info('skipping SHA-1 signature');
```

Comment:
We should make this fail instead.
sha1 should be removed from the configuration, so it's clear what we are doing. Silent failures are dangerous.

Reply:
Agree to disable it to explicit configuration, `electron-builder` does not provide a way to remove SHA-1: ["Windows is dual code-signed (SHA1 & SHA256 hashing algorithms)."](https://www.electron.build/code-signing.html). I think we can introduce an environment variable to explicitly disable the dual signing behavior (e.g., `WINDOWS_SIGNING_ELECTRON_BUILDER_DISABLE_DUAL_SIGNING`).

Reply:
See `signingHashAlgorithms` from https://www.electron.build/configuration/win.html

Reply:
Great, but I would like to have two different hash algorithms for different signing methods, for example, having [`sha-1`, `sha-256`] for USB stick while having [`sha-256`] only for GCP HSM.

Reply:
I deleted these code and still keep the dual-signing option, if people want to use GCP, they need to explicitly disable sha-1 in the `electron-builder` configuration.